plugins/woocommerce-blocks/assets/js/atomic/utils/blocks-registration-manager/template-change-detector.ts

```diff
@@ -29,17 +34,71 @@ export interface TemplateChangeDetectorObserver {
 export class TemplateChangeDetector implements TemplateChangeDetectorSubject {
 	private previousTemplateId: string | undefined;
 	private currentTemplateId: string | undefined;
-	private isPostOrPage: boolean;
+	private previousContentType: ContentType | undefined;
+	private currentContentType: ContentType | undefined;
+	private previousPageLocation = '';
+	private currentPageLocation = '';
 
 	private observers: TemplateChangeDetectorObserver[] = [];
 
 	constructor() {
-		this.isPostOrPage = false;
 		subscribe( () => {
-			this.checkIfTemplateHasChangedAndNotifySubscribers();
+			this.previousPageLocation = this.currentPageLocation;
+			this.currentPageLocation = window.location.href;
+			this.previousContentType = this.currentContentType;
+			this.currentContentType = this.detectContentType();
+			const isPageLocationUpdated = this.checkIfPageLocationHasChanged();
+
+			if ( isPageLocationUpdated ) {
+				this.notify();
+			}
 		}, 'core/edit-site' );
 	}
 
+	private detectContentType(): ContentType {
+		const path = getPath( window.location.href );
+
+		if ( ! path ) {
+			return ContentType.NONE;
+		}
+
+		if ( path.includes( 'site-editor.php' ) ) {
```

Comment:
You can edit page content from the Site Editor which would be incorrectly caught here from my understanding. The URL for editing pages in the Site Editor would look like this: `/wp-admin/site-editor.php?postType=page&postId=2`